plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/JavaToJKTreeBuilder.kt

```diff
@@ -904,6 +905,24 @@ class JavaToJKTreeBuilder(
                 it.withFormattingFrom(this)
             }
 
+        private fun PsiMethod.isOverrideMethodWithRedundantVisibility(): Boolean {
+            val superMethods = findSuperMethods()
+            if (superMethods.isEmpty()) return false // Unknown super method
+            val methodVisibility = visibility().visibility
+
+            // ensure none of the found super methods violate redundancy conditions
+            return superMethods.all { superMethod ->
+                val superVisibility = superMethod.visibility().visibility
+                when {
+                    name == "clone" && superMethod.containingClass?.name == "Object"
+                            && superMethods.size == 1 -> methodVisibility == PROTECTED // special built-in type
```

Comment:
A similar check also exists in `JKModifierElement.isRedundant`, is it necessary in both places?